### PR TITLE
get metaInfo from leader

### PR DESCRIPTION
**Pull request by Nicole00**

### CODE REVIEW

File: client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java [hunk 1]
```diff
@@ -137,10 +136,22 @@ public int getSpaceIdFromCache(String name) {
...
         ListSpacesReq request = new ListSpacesReq();
-        ListSpacesResp response;
+        ListSpacesResp response = null;
         try {
-            response = client.listSpaces(request);
+            while (retry-- >= 0) {
+                response = client.listSpaces(request);
+                if (response.code == ErrorCode.E_LEADER_CHANGED) {
+                    close();
+                    HostAddr newLeader = response.getLeader();
```
Comment (laura-ding) [inline client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java:147]: I think here is better to provide a function to return a new `metaClient`, and it's not only happening in `listSpaces`, other meta interfaces also do it too.

Comment (Nicole00) [reply client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java]: Great idea. Leader change will happen  at any time.

File: client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java [hunk 2]
```diff
@@ -137,14 +149,22 @@ public int getSpaceIdFromCache(String name) {
...
     public List<SpaceNameID> listSpaces() {
+
         ListSpacesReq request = new ListSpacesReq();
-        ListSpacesResp response;
+        ListSpacesResp response = null;
         try {
-            response = client.listSpaces(request);
+            while (retry-- >= 0) {
+                response = client.listSpaces(request);
+                freshClient(response.getCode(), response.getLeader());
```
Comment (laura-ding) [inline client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java:158]: here will do retry times whether it succeeds or fails. You need to jump out of the loop when succeeding.

Comment (Nicole00) [reply client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java]: right, stupid retry :(

File: client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java [hunk 3]
```diff
@@ -119,6 +117,18 @@ public int doConnect(List<HostAndPort> addresses) throws TException {
...
     }
 
+    private void getClient(String host, int port) throws TException {
+        transport = new TSocket(host, port, timeout, connectionTimeout);
+        transport.open();
+        protocol = new TCompactProtocol(transport);
+        client = new MetaService.Client(protocol);
+    }
+
+    private void freshClient(int responseCode, HostAddr leader) throws TException {
```
Comment (laura-ding) [inline client/src/main/java/com/vesoft/nebula/client/meta/MetaClientImpl.java:127]: the responseCode is no need

**Review by laura-ding — APPROVED** [review]

**Pull request merged**